runtime/lua/vim/lsp.lua

```diff
@@ -858,30 +858,19 @@ function lsp.omnifunc(findstart, base)
     end
   end
 
+  local line = vim.api.nvim_get_current_line()
+  local pos = vim.api.nvim_win_get_cursor(0)
+  local line_to_cursor = line:sub(1, pos[2])
```

Comment:
This would go up to, but not including the cursor, which potentially makes sense to me, but I want to confirm that's what you wanted.